### PR TITLE
fix: various site errors

### LINKED ISSUE #196: Assorted site errors
- [x] 87 x `PHP Notice:  Trying to access array offset on value of type bool in D:\home\site\wwwroot\_includes\kmdevhtml.php on line 789`
- [x] 57 x `PHP Notice:  Trying to access array offset on value of type null in D:\home\site\wwwroot\_includes\kmdevhtml.php on line 795`
- [x] 29 x `PHP Warning:  explode(): Empty delimiter in D:\home\site\wwwroot\_includes\kmdevhtml.php on line 788`
- [x] 29 x `PHP Warning:  count(): Parameter must be an array or an object that implements Countable in D:\home\site\wwwroot\_includes\kmdevhtml.php on line 800`
- [ ] 26 x `PHP Warning:  file_get_contents(): SSL operation failed with code 1. OpenSSL Error messages: error:14095126:SSL routines:ssl3_read_n:unexpected eof while reading in D:\home\site\wwwroot\_includes\includes\renderLanguageExample.php on line 25`
- [ ] 17 x `PHP Warning:  file_get_contents(): SSL: The operation completed successfully.  in D:\home\site\wwwroot\_includes\includes\renderLanguageExample.php on line 25`
- [x]  6 x `PHP Notice:  Undefined variable: site_tavultesoft in D:\home\site\wwwroot\keyboard\ekwunitamil\1.0\ekwunitamil.php on line 22`
- [x]  5 x `PHP Notice:  Undefined variable: site_tavultesoft in D:\home\site\wwwroot\keyboard\visual_media_tamil_typewriter\1.0\visual_media_tamil_typewriter.php on line 22`
- [x]  4 x `PHP Notice:  Undefined variable: site_tavultesoft in D:\home\site\wwwroot\keyboard\ekwnewtwuni\1.0\ekwnewtwuni.php on line 22`
- [x]  2 x `PHP Warning:  Invalid argument supplied for foreach() in D:\home\site\wwwroot\_includes\kmdevhtml.php on line 447`
- [x]  2 x `PHP Warning:  explode(): Empty delimiter in D:\home\site\wwwroot\_includes\kmdevhtml.php on line 445`
- [ ]  2 x `PHP Notice:  Undefined offset: 1 in D:\home\site\wwwroot\_includes\kmdevhtml.php on line 248`
- [ ]  1 x `PHP Notice:  Trying to access array offset on value of type null in D:\home\site\wwwroot\_includes\includes\renderLanguageExample.php on line 32`
- [ ]  1 x `PHP Notice:  Trying to access array offset on value of type null in D:\home\site\wwwroot\_includes\includes\renderLanguageExample.php on line 34`
- [x]  1 x `PHP Notice:  Undefined index: HTTP_USER_AGENT in D:\home\site\wwwroot\_includes\includes\template.php on line 21`
- [x]  1 x `PHP Notice:  Undefined index: HTTP_USER_AGENT in D:\home\site\wwwroot\_includes\includes\template.php on line 23`
- [x]  1 x `PHP Notice:  Undefined index: HTTP_USER_AGENT in D:\home\site\wwwroot\_includes\includes\template.php on line 25`
- [x]  1 x `PHP Notice:  Undefined index: HTTP_USER_AGENT in D:\home\site\wwwroot\_includes\includes\template.php on line 27`
- [x]  1 x `PHP Notice:  Undefined index: HTTP_USER_AGENT in D:\home\site\wwwroot\_includes\includes\template.php on line 29`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 106`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 113`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 120`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 127`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 134`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 141`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 148`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 29`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 36`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 43`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 50`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 57`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 64`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 71`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 78`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 85`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 92`
- [x] ​ 1 x `PHP Notice:  Undefined variable: site_keymanweb_static in D:\home\site\wwwroot\developer\engine\web\1.0\content\troubleshooting\embedded_fonts.php on line 99`
- [x]  1 x `PHP Notice:  Undefined variable: site_tavultesoft in D:\home\site\wwwroot\keyboard\ekwbamuni\1.0\ekwbamuni.php on line 22`
- [x]  1 x `PHP Notice:  Undefined variable: site_tavultesoft in D:\home\site\wwwroot\keyboard\ekwnewtwuni\1.0\chart.php on line 22`
- [x]  1 x `PHP Notice:  Undefined variable: site_tavultesoft in D:\home\site\wwwroot\keyboard\ekwnewtwuni\1.0\srii\tamil_srii.php on line 22`
- [x]  1 x `PHP Notice:  Undefined variable: site_tavultesoft in D:\home\site\wwwroot\keyboard\ekwtamil99uni\2.0.2\ekwtamil99uni.php on line 23`
- [ ]  1 x `PHP Warning:  file_get_contents(https://api.keyman.com/cloud/3.0/keyboards/chart): failed to open stream: HTTP request failed! HTTP/1.1 404 Failure in D:\home\site\wwwroot\_includes\includes\renderLanguageExample.php on line 25`
- [ ]  1 x `PHP Warning:  Invalid argument supplied for foreach() in D:\home\site\wwwroot\_includes\includes\renderLanguageExample.php on line 34`

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #196.

Note, not every error has been addressed from the issue. There are a few which are less obvious and for which I will wait until we start to get Sentry reports for them.

### CODE REVIEW

File: _includes/kmdevhtml.php [hunk 1]
```diff
@@ -39,14 +40,14 @@ function dispatcher( $func, $args ) {
       case "FEATURELIST": echo FEATURELIST( $args ); break;
       case "FEATURE": echo FEATURE( $args ); break;
       case "ENDFEATURELIST": echo ENDFEATURELIST( $args ); break;
-      
+
```
Comment (darcywong00) [inline _includes/kmdevhtml.php:43]: l. 18 can be removed (duplicate with 10)

File: developer/engine/web/1.0/content/troubleshooting/embedded_fonts.php [hunk 2]
```diff
@@ -1,164 +1,170 @@
...
-  
+
+use Keyman\Site\Common\KeymanHosts;
+
+require_once('servervars.php');
+
   $pagename = 'Embedded Font Troubleshooting';
   $pagetitle = 'Embedded Font Troubleshooting';
+
+  $site_keymanweb = KeymanHosts::Instance()->s_keyman_com;
```
Comment (darcywong00) [inline developer/engine/web/1.0/content/troubleshooting/embedded_fonts.php:26]: Should this be the s.keyman.com or keymanweb.com repo?

I see the path in l.35 at https://github.com/keymanapp/keymanweb.com/tree/master/fonts

whereas s.keyman.com would be `s.keyman.com/font/deploy/...`

File: _includes/kmdevhtml.php (hunk 1 above)
Comment (mcdurdin) [reply _includes/kmdevhtml.php:43]: Okay but I don't want to do other edits to this file ;-) It's super old!

File: developer/engine/web/1.0/content/troubleshooting/embedded_fonts.php (hunk 2 above)
Comment (mcdurdin) [reply developer/engine/web/1.0/content/troubleshooting/embedded_fonts.php:26]: This page is really old and doesn't work. Given it's for the obsolete version of KeymanWeb, I don't really want to spend time on it :)

**Pull request merged**